controllers/argocd_metrics_controller.go

```diff
@@ -282,6 +337,18 @@ func (r *ArgoCDMetricsReconciler) createServiceMonitorIfAbsent(namespace string,
 	return err
 }
 
+func (r *ArgoCDMetricsReconciler) deleteServiceMonitor(name string, namespace string, reqLogger logr.Logger) error {
+
+	err := r.Client.Delete(context.TODO(), &monitoringv1.ServiceMonitor{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: name}})
+	if err != nil {
+		reqLogger.Error(err, "Error deleting servicemonitor",
+			"Namespace", namespace)
+		return err
+	}
+	return nil
```

Comment:
Hi @saumeya, I just realized that return on not found error would lead to incomplete cleanup due to how the code is structured. Can we update the logic to below in all the places? This will ensure cleanup is proper if first resource is deleted successfully but subsequent resources are deleted in the next reconciliation cycle. 

```go
if !apierrors.IsNotFound(err) {
  reqLogger.Error(err, "Error deleting servicemonitor", "Namespace", namespace)
  return err
}
```